### PR TITLE
drivers: sensor: bmi160: Fix device definition on bus I2C

### PR DESCRIPTION
**Pull request by vinceWii4**
When the bmi160 device is located on bus i2c:
 - the anymotion trigger can't be mapped to int-pin
 - interrupt status register can't be read

Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

### CODE REVIEW

**Review by sjg20 — APPROVED** [review]
Not keen on the #if stuff but I don't have a better idea

**Comment by vinceWii4** [comment]
1. I updated _**drivers: sensor: bmi160: Fix device definition on bus I2C**_ commit message body
2. I added a commit to fix an issue in bmi160_trigger.c

File: drivers/sensor/bmi160/bmi160_trigger.c [hunk 1]
```diff
@@ -5,6 +5,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT bosch_bmi160
```
Comment (MaureenHelm) [inline drivers/sensor/bmi160/bmi160_trigger.c:8]: Please squash this change into the previous commit

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by galak** [comment]
@vinceWii4 Any updates.

**Comment by galak** [comment]
This also needs to be submitted to master branch first before we backport.

**Pull request closed without merging**